v2/yarpcgrpcfx/yarpcgrpcfx.go

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package yarpcgrpcfx
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/opentracing/opentracing-go"
+	"go.uber.org/config"
+	"go.uber.org/fx"
+	"go.uber.org/yarpc/v2"
+	"go.uber.org/yarpc/v2/yarpcgrpc"
+	"go.uber.org/zap"
+)
+
+const (
+	_name                     = "yarpcgrpcfx"
+	_inboundConfigurationKey  = "yarpc.grpc.inbounds"
+	_outboundConfigurationKey = "yarpc.grpc.outbounds"
+)
+
+// Module produces yarpcgrpc clients and starts yarpcgrpc inbounds.
+var Module = fx.Options(
+	fx.Provide(NewInboundConfig),
+	fx.Provide(NewOutboundsConfig),
+	fx.Provide(NewClients),
+	fx.Provide(NewDialer),
+	fx.Invoke(StartInbounds),
+)
+
+// InboundConfig is the configuration for starting yarpcgrpc inbounds.
+type InboundConfig struct {
+	Address string `yaml:"address"`
+}
+
+// InboundConfigParams defines the dependencies of this module.
+type InboundConfigParams struct {
+	fx.In
+
+	Provider config.Provider
+}
+
+// InboundConfigResult defines the values produced by this module.
+type InboundConfigResult struct {
+	fx.Out
+
+	Config InboundConfig
+}
+
+// NewInboundConfig produces an InboundConfig.
+func NewInboundConfig(p InboundConfigParams) (InboundConfigResult, error) {
+	ic := InboundConfig{}
+	if err := p.Provider.Get(_inboundConfigurationKey).Populate(&ic); err != nil {
+		return InboundConfigResult{}, err
+	}
+	return InboundConfigResult{
+		Config: ic,
+	}, nil
+}
+
+// StartInboundsParams defines the dependencies of this module.
+type StartInboundsParams struct {
+	fx.In
+
+	Lifecycle fx.Lifecycle
+	Router    yarpc.Router
```

Comment:
Nit: I think I updated this in a later PR, but what are your thoughts on separating the YARPC-specific dependencies from the rest?

```
type StartInboundsParams struct {
  fx.In

  Config InboundConfig
  Router yarpc.Router

  ...
}
```